### ISSUE TITLE
[Bug] [Steam] Achievements unlock erratically, unpredictably, and sometimes just don't at all.

### ISSUE THREAD

**Issue by hubismal** [Regalis11/Barotrauma#1394]
Title. Achievements are extremely strange, for me, and for everyone I've asked.

>Most achievements don't achieve mid round regardless of enablecheats (I've literally started up the game, created a new game, immediately gone to do the achievement, and it won't achieve.)

>Some achievements seem to only achieve in multiplayer games despite being singleplayer compatible. Some of the ones that only achieved in MP for me were the killing missions (moloch and hammerhead), The seafarer achivements (travel X kilometers, only achieved after that distance had been traveled in MP).

>Some achievements only seem to achieve after the round is over despite not being "complete a round after..." achievement. I can't recall which ones off the top of my head.

>Other achievements just seem to refuse unlocking. An example of this is Poisoner (kill something with morbusine) won't work at all.

>Some work just fine, at least for me. The CPR achievement and the Gotta Go Fast achievement worked just fine in a single player new game.

These conditions i've described could mean absolutely nothing; Some of the conditions are so weird it seems like they're almost random.
It should be noted that steam randomly crashes in the background sometimes for me. But the ones I am unable to achieve I have tried everything while steam is running. Additionally, other people report the same problems.

**Comment by PicolasCage** [comment]
I agree, pretty weird. For me the opiate addiction achievement unlocked ten minutes after actually recovering from them, in game.

**Comment by hubismal** [comment]
my inner completionist is clawing out my organs right now. i honestly hope this gets fixed before the actual netcode does lmao

**Comment by Regalis11** [comment]
Partially fixed in bd174a8 and 7ffcee3

**Comment by hubismal** [comment]
To give an update on this, "the lone sailor" doesn't seem to properly unlock for me.

**Comment by Regalis11** [comment]
I'm closing this for now. There could still be achievements that have similar problems, so please let me know if you run any problems and I'll reopen this.

**Comment by Razor94** [comment]
I have some issues with the killing a moloch achievement. Tried to unlock it in multiplayer and singleplayer, tried also ending the mission properly by docking at the destination port, nothing worked. I made sure i was the only gunner so i should have gotten the lasthit, too.  A friend of mine got the achievment instantly, while in the same session (i hosted), when he shot one during another mission. Like hubismal says, my inner completionist is tearing up.
